### PR TITLE
[5.4] Execute queries with lock only in write database

### PR DESCRIPTION
**Pull request by SebastianBerc**
As in PR #17317 but with the better solution and I added the test as a bug proof.

/cc @JoostK, @GrahamCampbell

### CODE REVIEW

**Comment by SebastianBerc** [comment]
I have no idea why StyleCI can't analysis this PR :/

File: src/Illuminate/Mail/Transport/ArrayTransport.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function messages()
     /**
      * Clear all of the messages from the local collection.
      *
-     * @return void
+     * @return Collection
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Transport/ArrayTransport.php]: Please don't make unrelated changes in the PR. This change also doesn't match our code standards anyway.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Mail/Transport/ArrayTransport.php (hunk 1 above)
Comment (SebastianBerc) [inline src/Illuminate/Mail/Transport/ArrayTransport.php]: I removed commit with that.
